drivers/sensor/wsen/wsen_pdus_25131308XXXXX/wsen_pdus_25131308XXXXX.c

```diff
@@ -0,0 +1,156 @@
+
+	switch (chan) {
+	case SENSOR_CHAN_ALL: {
+		if (PDUS_getRawPressureAndTemperature(&data->sensor_interface, &data->pressure,
+						      &data->temperature) != WE_SUCCESS) {
+			LOG_ERR("Failed to fetch data sample");
+			return -EIO;
+		}
+		break;
+	}
+	case SENSOR_CHAN_AMBIENT_TEMP: {
+		if (PDUS_getRawPressureAndTemperature(&data->sensor_interface, &pressure_dummy,
+						      &data->temperature) != WE_SUCCESS) {
+			LOG_ERR("Failed to fetch data sample");
+			return -EIO;
+		}
+		break;
+	}
+	case SENSOR_CHAN_PRESS: {
+		if (PDUS_getRawPressure(&data->sensor_interface, &data->pressure) != WE_SUCCESS) {
+			LOG_ERR("Failed to fetch data sample");
+			return -EIO;
+		}
+		break;
+	}
+	default:
+		LOG_ERR("Fetching is not supported on channel %d.", chan);
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int pdus_25131308XXXXX_channel_get(const struct device *dev, enum sensor_channel chan,
+					  struct sensor_value *value)
+{
+	struct pdus_25131308XXXXX_data *data = dev->data;
+	const struct pdus_25131308XXXXX_config *config = dev->config;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP: {
+		int32_t temperature_mega = ((int32_t)(data->temperature - T_MIN_VAL_PDUS)) * 4272;
+
+		value->val1 = temperature_mega / 1000000;
+		value->val2 = temperature_mega % 1000000;
+		break;
+	}
+	case SENSOR_CHAN_PRESS: {
+		int32_t pressure_temp = ((int32_t)(data->pressure - P_MIN_VAL_PDUS));
+
+		/*
+		 * these values are conversion factors based on the sensor type defined in the user
+		 * manual of the respective sensor
+		 */
+		switch (config->sensor_type) {
+		case PDUS_pdus0:
+			value->val1 = ((pressure_temp * 763) - 10000000) / 100000000;
+			value->val2 = (((pressure_temp * 763) - 10000000) % 100000000) / 100;
+			break;
+		case PDUS_pdus1:
+			value->val1 = ((pressure_temp * 763) - 10000000) / 10000000;
+			value->val2 = (((pressure_temp * 763) - 10000000) % 10000000) / 10;
+			break;
+		case PDUS_pdus2:
+			value->val1 = ((pressure_temp * 763) - 10000000) / 1000000;
+			value->val2 = ((pressure_temp * 763) - 10000000) % 1000000;
+			break;
+		case PDUS_pdus3:
+			value->val1 = (pressure_temp * 3815) / 1000000;
+			value->val2 = (pressure_temp * 3815) % 1000000;
+			break;
+		case PDUS_pdus4:
+			value->val1 = ((pressure_temp * 4196) - 10000000) / 100000;
+			value->val2 =
+				((pressure_temp * 4196) - 10000000) % 100000 * (1000000 / 100000);
+			break;
+		case PDUS_pdus5:
+			value->val1 = (pressure_temp * 5722) / 100000;
+			value->val2 = (pressure_temp * 5722) % 100000 * (1000000 / 100000);
+			break;
+		default:
+			LOG_ERR("Sensor type doesn't exist");
+			return -ENOTSUP;
+		}
+		break;
+	}
+	default:
+		LOG_ERR("Channel not supported %d", chan);
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api pdus_25131308XXXXX_driver_api = {
+	.sample_fetch = pdus_25131308XXXXX_sample_fetch,
+	.channel_get = pdus_25131308XXXXX_channel_get};
+
+static int pdus_25131308XXXXX_init(const struct device *dev)
```

Comment:
Should we try checking the Device ID before declaring a successful init? Similar to:
https://github.com/zephyrproject-rtos/zephyr/blob/ee250a11ce503fdf24793afac0712bacd26759da/drivers/sensor/wsen/wsen_hids_2525020210002/wsen_hids_2525020210002.c#L208-L212